imports/wasi_snapshot_preview1/fs.go

```diff
@@ -744,12 +744,87 @@ var pathCreateDirectory = stubFunction(
 // returns the attributes of a file or directory.
 //
 // See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-path_filestat_getfd-fd-flags-lookupflags-path-string---errno-filestat
-var pathFilestatGet = stubFunction(
-	functionPathFilestatGet,
-	[]wasm.ValueType{i32, i32, i32, i32, i32},
-	[]string{"fd", "flags", "path", "path_len", "result.buf"},
+var pathFilestatGet = &wasm.HostFunc{
+	ExportNames: []string{functionPathFilestatGet},
+	Name:        functionPathFilestatGet,
+	ParamTypes:  []api.ValueType{i32, i32, i32, i32, i32},
+	ParamNames:  []string{"fd", "flags", "path", "path_len", "result.buf"},
+	ResultTypes: []api.ValueType{i32},
+	Code: &wasm.Code{
+		IsHostFunction: true,
+		GoFunc:         api.GoModuleFunc(pathFilestatGetFn),
+	},
+}
+
+type wasiFiletype uint8
+
+const (
+	wasiFiletypeUnknown         wasiFiletype = 0
+	wasiFiletypeBlockDevice                  = 1
+	wasiFiletypeCharacterDevice              = 2
+	wasiFiletypeDirectory                    = 3
+	wasiFiletypeRegularFile                  = 4
+	wasiFiletypeSocketDgram                  = 5
+	wasiFiletypeSocketStream                 = 6
+	wasiFiletypeSymbolicLink                 = 7
 )
 
+func pathFilestatGetFn(ctx context.Context, mod api.Module, params []uint64) []uint64 {
+	fd := uint32(params[0])
```

Comment:
so I think here fd is what to use to lookup from, and path is the relative one to that.

ex. fd=3 for '/' and "foo/bar" as path, resulting in `path_filestat_get(3, "foo/bar"...)` returning stat on /foo/bar, not "/"

Reply:
I should probably have added some description in this PR.
`fstat` doesn't have these parameters so I didn't implement them. I think it's to support `fstat` and `fstatat` with one WASI call. I'm fine with adding support for these params, but for `fstat` the current implementation works.

Reply:
Roger. in any case this isn't about fstat it is about the wasi thing which is quasi similar. We won't be able to merge something that ignores its path param.. hope you understand https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-path_filestat_getfd-fd-flags-lookupflags-path-string---errno-filestat

Reply:
@codefromthecrypt See my last edit, I'll close this PR and keep it like it is for future reference, will implement `fd_filestat_get` now.